portality/view/openurl.py

```diff
@@ -0,0 +1,92 @@
+import re
+from flask import Blueprint, request, redirect, url_for, render_template, abort
+from portality.models import OpenURLRequest
+from urllib import unquote
+
+blueprint = Blueprint('openurl', __name__)
+
+@blueprint.route("/openurl", methods=["GET", "POST"])
+def openurl():
+
+    # Drop the first part of the url to get the raw query
+    url_query = request.url.split(request.base_url).pop()
+
+    # Validate the query syntax version and build an object representing it
+    parsed_object = parse_query(url_query)
+
+    # If it's not parsed to an OpenURLRequest, it's a redirect URL to try again.
+    if type(parsed_object) != OpenURLRequest:
+        return redirect(parsed_object, 301)
+
+    # Issue query and redirect to results page
+    results = parsed_object.query_es()
+    results_url = get_result_page(results)
+    if results_url:
+        return redirect(results_url)
+    else:
+        abort(400)
+
+def parse_query(url_query_string):
+    """
+    Create the model which holds the query
+    :param url_query_string: The query part of an incoming OpenURL request
+    :return: an object representing the query
+    """
+    # Check if this is new or old syntax, translate if necessary
+    match_1_0 = re.compile("url_ver=Z39.88-2004")
+    if not match_1_0.search(url_query_string):
+        print "Legacy OpenURL 0.1 request: " + unquote(request.url)
```

Comment:
app.logger.info or .warn or .error or even .debug may be best. from portality.core import app if not imported to get access to the logger. It'll give the message a timestamp too.

Also URL might contain unicode, I think? Probably good to take care of that here. If it's already a decoded string (so not a unicode obj) which it might well be (need to see Flask docs), then we can print safely.